### ISSUE TITLE
How to Call function entity By passing entity Object Custom 

### ISSUE THREAD

**Issue by ramireddyraam** [object/Simple.OData.Client#289]
i need to create request like 
```
POST https://ebiztest.aoa.org/PersonifyDataServices/PersonifyDataAoaCustom.svc/AddCustomerCreditCard HTTP/1.1
Content-Type: application/xml;charset=utf-8
ActivityId: WPF7WML
Authorization: Basic YW9hX3NzbzpTdDM0bTNkSDRtJA==
Host: ebiztest.aoa.org
Content-Length: 716
Expect: 100-continue

<?xml version="1.0" encoding="utf-16"?>
<CustomerCreditCardInput xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <InternalKey />
  <NavigationKey />
  <MasterCustomerId>00101247</MasterCustomerId>
  <SubCustomerId>0</SubCustomerId>
  <ReceiptType>Visa</ReceiptType>
  <CreditCardNumber>4111111111111111</CreditCardNumber>
  <ExpirationMonth>11</ExpirationMonth>
  <ExpirationYear>2020</ExpirationYear>
  <Cvv2>123</Cvv2>
  <NameOnCard>raja</NameOnCard>
  <BillingAddressStreet />
  <BillingAddressCity />
  <BillingAddressState />
  <BillingAddressPostalCode />
  <BillingAddressCountryCode />
  <AddedOrModifiedBy />
</CustomerCreditCardInput>
```
**ENTITY**
```
<FunctionImport Name="AddCustomerCreditCard" ReturnType="Edm.String" m:HttpMethod="POST">
<Parameter Name="CustomerCreditCardInput" Type="Edm.String" Mode="In"/>
</FunctionImport>
```

**Comment by object** [comment]
For functions and actions please have a look at examples here:

https://github.com/object/Simple.OData.Client/blob/master/Simple.OData.Client.IntegrationTests/TripPinTests.cs

(search for "Function" or "Action" in the code).

Here are some examples for OData V3:
https://github.com/object/Simple.OData.Client/blob/master/Simple.OData.Client.Tests.Net40/FunctionTests.cs

**Comment by ramireddyraam** [comment]
Hi Vagif Abilov

i have One function(

> AddCustomerCreditCard

) which accept `

>  CustomerCreditCardInput

`Entity and which return 

> CustomerCreditCardOutput


Here is odata service url : https://ebiztest.aoa.org/PersonifyDataServices/PersonifyDataAoaCustom.svc
user = "aoa_sso";
password = "St34m3dH4m$";

Function name : AddCustomerCreditCard
Input Entity : CustomerCreditCardInput
OutPut Entity : CustomerCreditCardOutput

Please suggest how to execute  those entities

**Comment by object** [comment]
I will check this out and come back to you later.

**Comment by ramireddyraam** [comment]
Thank you

**Comment by ramireddyraam** [comment]
i am waiting for your replay

**Comment by object** [comment]
Sorry this is not my main job project, I'll come back to you as soon as I can.

**Comment by object** [comment]
Hi, I tried this now but am getting an exception when executing the HTTP request. Will look into it tonight.

**Comment by ramireddyraam** [comment]
Ok thanks 

can you give that code i'll also try

**Comment by object** [comment]
Here's the example:

class CustomerCreditCardInput
{
    public string MasterCustomerId { get; set; }
    public int SubCustomerId { get; set; }
    public string ReceiptType { get; set; }
    public string CreditCardNumber { get; set; }
    public short ExpirationMonth { get; set; }
    public short ExpirationYear { get; set; }
    public string Cvv2 { get; set; }
    public string NameOnCard { get; set; }
}
 
[Fact]
public async Task IssueFunction()
{
    var client = new ODataClient(new ODataClientSettings()
    {
        UrlBase = "https://ebiztest.aoa.org/PersonifyDataServices/PersonifyDataAoaCustom.svc",
        Credentials = new NetworkCredential(..removed_credentials...),
    });
 
    var result = await client
        .Unbound()
        .Action("AddCustomerCreditCard")
        .Set(
            new CustomerCreditCardInput()
            {
                MasterCustomerId = "00101247",
                SubCustomerId = 0,
                ReceiptType = "VISA",
                CreditCardNumber = "4111111111111111",
                ExpirationMonth = 11,
                ExpirationYear = 2020,
                Cvv2 = "123",
                NameOnCard = "raja"
            })
        .ExecuteAsScalarAsync<string>();
}

**Comment by object** [comment]
Hi again,
I can confirm that this is a bug in a current version (failure to serialize action parameters for OData V3). I will let you know when it's fixed.

**Comment by ramireddyraam** [comment]
Hi 

I have another doubt on set attribute 
how to send dynamic  string or Object

**Comment by object** [comment]
Let us come back to it when I fix the bug. I have identified the problem and should be able to have a fix within a few days.

**Comment by object** [comment]
The fix will be released as a part of 4.24.

**Comment by object** [comment]
Should work in 4.24.
